cpp/include/cudf/detail/utilities/device_operators.cuh

```diff
@@ -257,4 +257,92 @@ struct DeviceLeadLag {
   explicit CUDF_HOST_DEVICE inline DeviceLeadLag(size_type offset_) : row_offset(offset_) {}
 };
 
+/**
+ * @brief Binary bitwise `AND` operator
+ */
+struct DeviceBitAnd {
+  template <typename T, std::enable_if_t<std::is_integral_v<T>>* = nullptr>
+  CUDF_HOST_DEVICE inline T operator()(T const& lhs, T const& rhs) const
+  {
+    return lhs & rhs;
+  }
+
+  template <typename T, std::enable_if_t<std::is_integral_v<T>>* = nullptr>
+  CUDF_HOST_DEVICE static constexpr T identity()
+  {
+    if constexpr (std::is_same_v<T, bool>) {
+      return true;
+    } else {
+      return ~T{0};
+    }
+  }
+
+  template <typename T, std::enable_if_t<!std::is_integral_v<T>>* = nullptr>
+  CUDF_HOST_DEVICE static constexpr T identity()
+  {
+#ifndef __CUDA_ARCH__
+    CUDF_FAIL("Bitwise AND is only supported for integral types.");
+#else
+    CUDF_UNREACHABLE("Bitwise AND is only supported for integral types.");
+#endif
+    return T{};
```

Comment:
To clarify:  The choice to use `std::enable_if_t` instead of `CUDF_ENABLE_IF` here:  Was the choice made for consistency with the rest of the code?

Reply:
Yes I intentionally use it to be consistent with other structs in the code.